fish-rust/src/common.rs

```diff
@@ -687,80 +776,6 @@ fn unescape_string_internal(input: &wstr, flags: UnescapeFlags) -> Option<WStrin
     Some(result)
 }
 
-/// Reverse the effects of `escape_string_url()`. By definition the input should consist of just
-/// ASCII chars.
-fn unescape_string_url(input: &wstr) -> Option<WString> {
-    let mut result: Vec<u8> = Vec::with_capacity(input.len());
-    let mut i = 0;
-    while i < input.len() {
-        let c = input.char_at(i);
-        if c > '\u{7F}' {
-            return None; // invalid character means we can't decode the string
-        }
-        if c == '%' {
-            let c1 = input.char_at(i + 1);
-            if c1 == '\0' {
-                return None;
-            } else if c1 == '%' {
-                result.push(b'%');
-                i += 1;
```

Comment:
This doesn't correspond to the escape code - a `%` is encoded as `%25` and not `%%`.